Ordering of tags in result of taglist call.

dncnmcdougall opened the pull request:
using :tselect the tags are ordered in a nice way. Using "taglist" tey are not. This allows the buffer name to be specified to "taglist" which makes it prioritise the tags like :tselect would if you were n the buffer with the name specified.

Commit:
Extends "taglist" to provide optional tag ordering.

dncnmcdougall commented:
Forgive my audacity:
@brammool, @shougo, @justinmk

What is needed to progress this pull request?

justinmk commented:
@dncnmcdougall Wait for some feedback, maybe write a few more tests. If no one opposes it, it will get merged. If it gets ignored, rebase it and send a reminder every week :>

dncnmcdougall commented:
Excellent, thank you.

brcolow commented on runtime/doc/eval.txt:
```diff
@@ -7643,8 +7644,11 @@ tagfiles()	Returns a |List| with the file names used to search for tags
-taglist({expr})							*taglist()*
+taglist({expr}[, {buffername}])						*taglist()*
 		Returns a list of tags matching the regular expression {expr}.
+                Uses {buffername} to prioratise the results in the same was
```
prioratise => prioritize

brcolow replied:
was => way

brcolow commented on runtime/doc/eval.txt:
```diff
@@ -7643,8 +7644,11 @@ tagfiles()	Returns a |List| with the file names used to search for tags
 		Returns a list of tags matching the regular expression {expr}.
+                Uses {buffername} to prioratise the results in the same was
+                that |:tselect| does. See |tag-priority|. If {buffername} is
+                ommitted the tags are not prioritised.
```
prioritised => prioritized

dncnmcdougall commented:
@brcolow thank you for the comments. I have made these changes.

dncnmcdougall commented:
I have updated this to be in sync with master again. With help I found a cleaner way to do this. 
I do not understand why the test coverage has reduced, as there is a test. Perhaps this is because of the additions to the documentation? Or perhaps the test I added is not registered as a test with the system?

vim-ml commented:
Hi Duncan,

2016-12-7(Wed) 17:10:17 UTC+9 Duncan McDougall:
> I have updated this to be in sync with master again. With help I found a cleaner way to do this.
> 
> I do not understand why the test coverage has reduced, as there is a test.. Perhaps this is because of the additions to the documentation? Or perhaps the test I added is not registered as a test with the system?

Yes.
Your test is not running.

See src/testdir/README.txt

You are still only implementing 1).
You need to implement 2) or later.

Note:
Typo in 2).
s/.vim/.res/

--
Best regards,
Hirohito Higashi (a.k.a. h_east)

dncnmcdougall commented:
Thank you @vim-ml, I have added the test to the makefiles.

dncnmcdougall commented:
This is a small change which exposes a property that already exists to the scripting environment. 
The exposed property defaults to what it was before exposed, so that this is backwards compatible.

The functionality it adds is it allows users to write scripts that use tags in a more context aware way. 

What is still impeding the integration of this code?

chrisbra commented:
You need to be patient. It may take a while until Bram comes to it.

dncnmcdougall commented:
Right, yes of course. Thank you.

dncnmcdougall commented:
Re-based and merged.

dncnmcdougall commented:
Re-based and merged.

dncnmcdougall commented:
Re-based and merged.

dncnmcdougall commented:
Re-based and merged.
